### ISSUE TITLE
Add option to plot weight visualization during training

### ISSUE THREAD

**Issue by thomasZen** [open-retina/open-retina#140]
An option to implement it is the following:

I'd probably implement it as a pytorch lightning callback:
https://lightning.ai/docs/pytorch/stable/extensions/callbacks.html
So one solution is to create a new class inheriting from Callback and then overwriting for example the [on_validation_epoch_end](https://lightning.ai/docs/pytorch/stable/extensions/callbacks.html) function. You probably get the epoch number from the trainer argument of that function and the neural network based on the pl_module argument, and can then make it configurable to visualize the weights each x epochs.

We could even make it configurable what to visualize (ie core, readout, MEIs, but maybe that's overkill).

**Comment by thomasZen** [comment]
@openhands Create a PR for this.

**Comment by BaptisteLorenzi** [comment]
Hi, I wanted it and did it using pytorch lightning callback. It works well. One question is where to put those graphs. For now I'm using openretina_assets/runs/{my_exp}/model_weights/ . They could also be saved as MLFlow artefacts or in tensorboard for that matter. Finally I've been able to access the current epoch to use it the plot name.

**Comment by thomasZen** [comment]
> Hi, I wanted it and did it using pytorch lightning callback. It works well. One question is where to put those graphs. For now I'm using openretina_assets/runs/{my_exp}/model_weights/ . They could also be saved as MLFlow artefacts or in tensorboard for that matter. Finally I've been able to access the current epoch to use it the plot name.

I think `openretina_assets/runs/{my_exp}/weight_visualizations/` or `../model_visualizations` sounds good (in model_weights I would probably expect the binary weights or the weights as a checkpoint).

Wrt, MLFlow/tensorboard, I would probably make that a config option or skip for now, depending how difficult it is to get right. I think the advantage to just writing them in a folder is that it will work independant if you use mlflow, weights&biases, tensorboard, etc., so I would use that as the base.

@BaptisteLorenzi If you can raise that or your current state as a separate PR at some point when you have time it would be awesome. I'm also interested in how the model weights develop during training :) 

btw, also sorry for the openhands spam, I configured this at some point for me, but did not spend the time to make it actually useful.

**Comment by BaptisteLorenzi** [comment]
Hello, I Raised the current state of the callback in the goldin_2022 branch. I believe it needs a bit more work to work with all the models.

**Comment by thomasZen** [comment]
Added and merged with https://github.com/open-retina/open-retina/pull/150.
